kubetest2/internal/deployers/eksapi/k8s.go

```diff
@@ -9,35 +9,55 @@ import (
 	"time"
 
 	"github.com/aws/aws-k8s-tester/kubetest2/internal/metrics"
+	"github.com/aws/aws-k8s-tester/kubetest2/internal/util"
 	"github.com/aws/aws-sdk-go-v2/service/ec2"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	crlog "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
 	corev1 "k8s.io/api/core/v1"
 )
 
-func newKubernetesClient(kubeconfigPath string) (*kubernetes.Clientset, error) {
-	c, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+func init() {
+	// controller-runtime will complain loudly if this isn't set, even though we don't use this logger
+	crlog.SetLogger(zap.New())
+}
+
+type k8sClient struct {
+	config    *rest.Config
+	clientset kubernetes.Interface
+	client    client.Client
+}
+
+func newK8sClient(kubeconfigPath string) (*k8sClient, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
 	if err != nil {
 		return nil, err
 	}
-	return kubernetes.NewForConfig(c)
+	return &k8sClient{
+		config:    config,
+		clientset: kubernetes.NewForConfigOrDie(config),
+		client:    util.Must(client.New(config, client.Options{})),
```

Comment:
is this the only place the `Must` got used 🤣

Reply:
for now 😉